live/client_test.py

```diff
@@ -27,203 +34,174 @@ def mock_response(status=200, content="CONTENT",
 
 
 def test_ElementalLive_should_receive_server_ip():
-    e = ElementalLive('http://elemental.dev.cbsivideo.com/')
-    assert e.server_ip == 'http://elemental.dev.cbsivideo.com/'
+    e = ElementalLive(ELEMENTAL_ADDRESS)
+    assert e.server_ip == ELEMENTAL_ADDRESS
 
 
-@mock.patch('requests.request')
-def test_create_event_with_authentication(mock_request):
-    """test create_event method"""
-    response_from_elemental_api = open(
-        'live/templates/sample_response_for_create.xml').read()
-    mock_request.return_value = mock_response(
-        status=201, content=response_from_elemental_api)
+def test_generate_header_with_authentication_should_contain_user():
+    """"test generate function"""
+    client = ElementalLive(ELEMENTAL_ADDRESS, USER, API_KEY)
+    headers = client.generate_headers(f'{ELEMENTAL_ADDRESS}/live_events')
+    assert headers['X-Auth-User'] == USER
+    assert headers['Accept'] == 'application/xml'
+    assert headers['Content-Type'] == 'application/xml'
 
-    client = ElementalLive("http://elemental.dev.cbsivideo.com", USER, API_KEY)
-    response = client.create_event("live/templates/qvbr_mediastore.xml",
-                                   {'username': os.getenv('ACCESS_KEY'),
-                                    'password': os.getenv('SECRET_KEY'),
-                                    'mediastore_container_master':
-                                        'https://hu5n3jjiyi2jev.data.media'
-                                        'store.us-east-1.amazonaws.com/master',
-                                    'mediastore_container_backup':
-                                        'https://hu5n3jjiyi2jev.data.medias'
-                                        'tore.us-east-1.amazonaws.com/backup',
-                                    'channel': "1", 'device_name': "0"})
+
+def test_genterate_header_without_authentication_should_not_contain_user():
+    """"test generate function"""
+    client = ElementalLive(ELEMENTAL_ADDRESS)
+    headers = client.generate_headers()
+    assert 'X-Auth-User' not in headers
+    assert headers['Accept'] == 'application/xml'
+    assert headers['Content-Type'] == 'application/xml'
+
+
+@mock.patch('requests.request')
+def test_send_request_should_call_request_as_expected(mock_request):
+    """test send request method
+    to call request method"""
+    client = ElementalLive(ELEMENTAL_ADDRESS, USER, API_KEY)
+    mock_request.return_value = mock_response(status=200)
```

Comment:
Try to swap these two lines: mock first.
Then you create your objects and call the method being tested, then asserts. It's a common convention in tests

also I don't see a need for docstrings when the test name is already very descriptive